src/main/java/ch/interlis/ioxwkf/shp/ShapeWriter.java

```diff
@@ -652,20 +661,49 @@ public void setModel(TransferDescription td) {
 	}
 	
 	public AttributeDescriptor[] getAttributeDescriptors() {
-		return attrDescs.toArray(new AttributeDescriptor[attrDescs.size()]);
+        List<AttributeDescriptor> attrDescs = new ArrayList<AttributeDescriptor>();
+        for (Map.Entry<String, AttributeDescriptor> entry : attrDescsMap.entrySet()) {
+            attrDescs.add(entry.getValue());
+        }
+        return attrDescs.toArray(new AttributeDescriptor[attrDescs.size()]);
 	}
 	
 	/** Sets the attribute descriptors of the shape file.
 	 * Alternative to setting a model.
 	 * @param attrDescs[]
 	 */
 	public void setAttributeDescriptors(AttributeDescriptor attrDescs[]) {
```

Comment:
Das funktioniert so nicht via Db2Shp. M.E. muss hier auch gekürtzt werden.

Reply:
D.h. der fehlende Db2Shp-Test würde failen?

Reply:
Ja, hat gefailed. Trimmen wird jetzt auch in `setAttributeDescriptors` gemacht.